cli/src/semgrep/metrics.py

```diff
@@ -308,6 +316,15 @@ def add_findings(self, findings: FilteredMatches) -> None:
         self.payload.value.numFindings = sum(len(v) for v in findings.kept.values())
         self.payload.value.numIgnored = sum(len(v) for v in findings.removed.values())
 
+        # Breakdown # of findings per-product.
+        _num_findings_by_product: Dict[out.Product, int] = defaultdict(int)
+        for r, f in findings.kept.items():
```

Comment:
Unrelated to your PR but what is "kept"?

Reply:
"Kept" here is the opposite of "ignored". So it stores whatever matches that weren't ignored. If I understand correctly, when [the user uses `#nosemgrep`](https://github.com/semgrep/semgrep/blob/986e7772c6dfe1f174e919ad8fa0d6f4749a4c85/cli/src/semgrep/nosemgrep.py#L4-L5), that match will be ignored.